### PR TITLE
chore: tracing over bidi+

### PR DESCRIPTION
**Pull request by OrKoN**

### CODE REVIEW

File: packages/puppeteer-core/src/common/Page.ts [hunk 1]
```diff
@@ -184,7 +184,26 @@ export class CDPPage extends Page {
...
+      close: opts => {
+        return this.#client.send('IO.close', opts);
+      },
+      start: opts => {
+        return client.send('Tracing.start', opts);
+      },
+      stop: async () => {
+        const deferred = Deferred.create();
+        this.#client.once('Tracing.tracingComplete', event => {
+          console.log('resolved');
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/common/Page.ts]:
```suggestion

```

File: packages/puppeteer-core/src/common/bidi/BrowsingContext.ts [hunk 2]
```diff
@@ -289,6 +294,7 @@ export class BrowsingContext extends EventEmitter {
       context: this.#id,
     });
     const sessionId = session.result.cdpSession;
+    this.#cdpSessionId = sessionId;
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/common/bidi/BrowsingContext.ts]: Should we skip the first call if we have a `cdpSessionId` set?

Comment (OrKoN) [reply packages/puppeteer-core/src/common/bidi/BrowsingContext.ts]: I am not sure about the lifetime of sessions on the mapper side? perhaps we could

File: packages/puppeteer-core/src/common/Tracing.ts [hunk 3]
```diff
@@ -29,6 +32,11 @@ export interface TracingOptions {
   categories?: string[];
 }
 
+interface TracingSource extends ProtocolReadable {
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/common/Tracing.ts]: Please for the new interfaces provide `@internal` comment.

**Review by Lightning00Blade — APPROVED** [review]

**Pull request merged**